flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaDerivation.java

```diff
@@ -84,34 +88,50 @@ public List<SchemaChangeEvent> applySchemaChange(SchemaChangeEvent schemaChangeE
                 SchemaChangeEvent derivedSchemaChangeEvent =
                         ChangeEventUtils.recreateSchemaChangeEvent(schemaChangeEvent, derivedTable);
                 schemaManager.applySchemaChange(derivedSchemaChangeEvent);
-                return Collections.singletonList(derivedSchemaChangeEvent);
-            }
-
-            // Many-to-1 mapping (merging tables)
-            Schema derivedTableSchema = schemaManager.getLatestSchema(derivedTable).get();
-            if (schemaChangeEvent instanceof CreateTableEvent) {
-                return handleCreateTableEvent(
-                        (CreateTableEvent) schemaChangeEvent, derivedTableSchema, derivedTable);
-            } else if (schemaChangeEvent instanceof AddColumnEvent) {
-                return handleAddColumnEvent(
-                        (AddColumnEvent) schemaChangeEvent, derivedTableSchema, derivedTable);
-            } else if (schemaChangeEvent instanceof AlterColumnTypeEvent) {
-                return handleAlterColumnTypeEvent(
-                        (AlterColumnTypeEvent) schemaChangeEvent, derivedTableSchema, derivedTable);
-            } else if (schemaChangeEvent instanceof DropColumnEvent) {
-                return Collections.emptyList();
-            } else if (schemaChangeEvent instanceof RenameColumnEvent) {
-                return handleRenameColumnEvent(
-                        (RenameColumnEvent) schemaChangeEvent, derivedTableSchema, derivedTable);
+                events.add(derivedSchemaChangeEvent);
```

Comment:
```
// 1-to-1 mapping. Replace the table ID directly
```
This should be changed to include `1 to many`.

Reply:
I think "single source" might be more concise since route sink can be 1 or many.